benchmarks/src/generated/main/grpc/io/grpc/benchmarks/proto/BenchmarkServiceGrpc.java

```diff
@@ -38,7 +43,12 @@ private BenchmarkServiceGrpc() {}
               io.grpc.benchmarks.proto.Messages.SimpleRequest.getDefaultInstance()))
           .setResponseMarshaller(io.grpc.protobuf.ProtoUtils.marshaller(
               io.grpc.benchmarks.proto.Messages.SimpleResponse.getDefaultInstance()))
+          .setMethodOptions(SERVICE_PROTO_DESCRIPTOR
+              .getMethodDescriptor(METHODIDX_UNARY_CALL).getOptions())
```

Comment:
> the indexes should remain the same - if they're different it'll mean the files are different

Exactly. The files may be different. This is "okay" as long as the changes are backward-compatible. Adding a new method (in any order) is considered backward-compatible. But inserting a method like that would change the indexes.

Even if we were using the indexes directly, I'd want some sort of verification to fail when the index no longer matches. Otherwise it would be really painful to track down bugs caused by this.